source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,620 @@
+                               enet_event_t event,
+                               enet_frame_info_t *frameInfo,
+                               void *userData );
+
+static void prvProcessFrame( int length );
+
+BaseType_t xNetworkInterfaceInitialise( void )
+{
+	status_t xStatus;
+	phy_config_t xConfig = { 0 };
+	BaseType_t xResult = pdPASS;
+	phy_speed_t speed;
+    phy_duplex_t duplex;
+    status_t readStatus;
+    enet_config_t config;
+    uint32_t sysClock;
+    enet_buffer_config_t buffCfg[ENET_RING_NUM];
+    int i;
+    uint8_t ucCounter = 0;
+    bool autoNegotiationComplete = false;
+    bool LinkUp = false;
+
+    configASSERT( FSL_FEATURE_ENET_QUEUE == 1 );
+
+    #if defined( ENET_ENHANCEDBUFFERDESCRIPTOR_MODE )
+        configASSERT( pdFALSE );
+	#endif
+
+    ethernetifLocal->RxBuffDescrip = &( rxBuffDescrip_0[ 0 ] );
+    ethernetifLocal->TxBuffDescrip = &( txBuffDescrip_0[ 0 ] );
+    ethernetifLocal->RxDataBuff    = &( rxDataBuff_0[ 0 ] );
+    ethernetifLocal->TxDataBuff    = &( txDataBuff_0[ 0 ] );
+    ethernetifLocal->base          = ( void * ) ENET_BASE;
+
+	/* Set the clock frequency. */
+	mdioHandle.resource.csrClock_Hz = EXAMPLE_CLOCK_FREQ;
+	mdioHandle.resource.base = ( void * ) ENET_BASE;
+
+	/* Set the configuration to auto-negotiate; set the phy to full duplex mode; the phy's address; and the speed. */
+	xConfig.autoNeg = pdTRUE;
+	xConfig.duplex = kPHY_FullDuplex;
+	xConfig.phyAddr = BOARD_ENET0_PHY_ADDRESS;
+	xConfig.speed = kPHY_Speed100M;
+
+	configPRINTF( ( "Starting PHY initialization.\r\n" ) );
+
+	xStatus = PHY_Init( &phyHandle, &xConfig );
+
+	if( xStatus != kStatus_Success )
+	{
+		xResult = pdFAIL;
+		configPRINTF( ( "Failed to initialize the PHY..." ) );
+	}
+	else
+	{
+		configPRINTF( ( "Configuration successful. Waiting for autonegotiation to complete...\r\n" ) );
+
+		do
+		{
+			readStatus = PHY_GetAutoNegotiationStatus( &phyHandle, &autoNegotiationComplete );
+
+			if( autoNegotiationComplete == true )
+			{
+				break;
+			}
+
+			/* Try for only a limited number of times. */
+			if( ucCounter++ > MAX_AUTONEG_FAILURE_COUNT )
+			{
+				break;
+				xResult = pdFAIL;
+			}
+
+			vTaskDelay( pdMS_TO_TICKS( 10 ) );
+		}
+		while( readStatus == kStatus_Success );
+	}
+
+
+	if( xResult == pdPASS )
+	{
+		if( autoNegotiationComplete == false )
+		{
+			configPRINTF( ( "Failed to complete auto-negotiation. Using default values." ) );
+
+			xConfig.autoNeg = pdFALSE;
+
+			xStatus = PHY_Init( &phyHandle, &xConfig );
+			if( xStatus != kStatus_Success )
+			{
+				xResult = pdFAIL;
+				configPRINTF( ( "Failed to re-initialize the PHY..." ) );
+			}
+			else
+			{
+				configPRINTF( ( "PHY re-initialization successful." ) );
+			}
+		}
+		else
+		{
+			configPRINTF( ( "Auto-negotiation complete.\r\n" ) );
+		}
+
+		ucCounter = 0;
+		do
+		{
+			readStatus = PHY_GetLinkStatus( &phyHandle, &LinkUp );
+
+			if( LinkUp == true )
+			{
+				break;
+			}
+
+			vTaskDelay( pdMS_TO_TICKS( 1000 ) );
```

Comment:
Calling vTaskDelay() is no problem, as there is only one interface. In the /multi library you could return pdFAIL and wait to be called back again.